### PR TITLE
feat: upgrades to pydantic version 2

### LINKED ISSUE #53: Unpin pydantic version
# User story
As a user, I want to use different versions of pydantic, so I can install what is required.

# Acceptance criteria
- [ ] Unpin pydantic

# Sprint Ready Checklist 
1. - [ ] Acceptance criteria defined 
2. - [ ] Team understands acceptance criteria 
3. - [ ] Team has defined solution / steps to satisfy acceptance criteria 
4. - [ ] Acceptance criteria is verifiable / testable 
5. - [ ] External / 3rd Party dependencies identified
6. - [ ] Ticket is prioritized and sized

# Notes
*Add any helpful notes here.*

### PR DESCRIPTION
**Pull request by jtyoung84**
Closes #53 

- Updates pydantic to > version 2
- Had to refactor most of the Credentials Settings stuff

### CODE REVIEW

File: src/aind_codeocean_api/credentials.py [hunk 1]
```diff
@@ -1,46 +1,137 @@
...
+    def get_field_value(
+        self, field: FieldInfo, field_name: str
+    ) -> Tuple[Any, str, bool]:
+        """This function needs to be implemented for every
+        PydanticBaseSettingsSource"""
+        file_content_json = self._json_contents
+        field_value = file_content_json.get(field_name)
+        return field_value, field_name, False
+
+    def prepare_field_value(
```
Comment (camilolaiton) [inline src/aind_codeocean_api/credentials.py:56]: What is the purpose of having this method if it only returns the value?

Comment (jtyoung84) [reply src/aind_codeocean_api/credentials.py:56]: I followed this example from the pydantic documentation directly. I think these methods need to be implemented.

File: src/aind_codeocean_api/credentials.py [hunk 2]
```diff
@@ -1,46 +1,137 @@
...
+        self,
+        field_name: str,
+        field: FieldInfo,
+        value: Any,
+        value_is_complex: bool,
+    ) -> Any:
+        """This function needs to be implemented for every
```
Comment (camilolaiton) [inline src/aind_codeocean_api/credentials.py:63]: There are some methods that don't have the parameters or returns in the docs and these won't be reflected in the package documentation site.

File: src/aind_codeocean_api/credentials.py [hunk 3]
```diff
@@ -1,46 +1,137 @@
...
+
+class AWSConfigSettingsSource(JsonConfigSettingsSource):
+    """Class that parses from aws secrets manager."""
+
+    @staticmethod
+    def _get_secret(secret_name: str) -> dict:
+        """
+        Retrieves a secret from AWS Secrets Manager.
+
+        param secret_name: The name of the secret to retrieve.
```
Comment (camilolaiton) [inline src/aind_codeocean_api/credentials.py:103]: Follow the same documentation standard

**Review by camilolaiton — APPROVED** [review]
Everything looks good to me, only a couple comments about docs

File: src/aind_codeocean_api/credentials.py (hunk 2 above)
Comment (jtyoung84) [reply src/aind_codeocean_api/credentials.py]: Updated doc strings

File: src/aind_codeocean_api/credentials.py (hunk 3 above)
Comment (jtyoung84) [reply src/aind_codeocean_api/credentials.py]: Updated format

**Pull request merged**